src/game/Spells/Spell.cpp

```diff
@@ -3064,6 +3064,17 @@ SpellCastResult Spell::SpellStart(SpellCastTargets const* targets, Aura* trigger
         return SPELL_FAILED_SPELL_IN_PROGRESS;
     }
 
+    // There are some items that do not reliably filter client-side if they are usable 
```

Comment:
Checks like this belong to Spell::CheckCast and generally should use the "cycle" with MAX_EFFECT_INDEX paradigm instead of 3 hardcoded checks.